### ISSUE TITLE
Deploy build directory to github pages?

### ISSUE THREAD

**Issue by rpbaptist** [middleman-contrib/middleman-deploy#112]
Obviously I am overlooking something, but I fail to understand how the deploy command works. It pushes the entire middleman repo, but I would expect it to push the `build` directory only as that is the actual site content.

How do I configure this so that the build directory is deployed to my github pages repo instead of the entire middleman repo? Or do I need to specify github pages to somehow only load up files from the `build` directory?

**Comment by rpbaptist** [comment]
Figured it out. I had build in my `.gitignore` from before. After I changed it, it behaves exactly as expected. Very useful.

**Comment by rpbaptist** [comment]
Actually it still doesn't work. I have now removed middleman-deploy and set my github pages as a remote for the build directory and wrote my own script.

**Comment by hello-jason** [comment]
I've been using this gem for a long time without issues. Only very recently am I having the same issue as you, where it deploys the entire project to the gh-pages branch rather than just the build directory.

I'll post back if I figure out what's going on.

**Comment by juanpabloe** [comment]
@rpbaptist could you share the script you wrote?
@hello-jason same thing happening here

**Comment by rpbaptist** [comment]
The problem (or design) is that the entire `build` directory is removed. It is impossible to keep this as a separate git repo.

I figured two ways to do this. One is to point the root of your website to `http://example.com/build`, but I didn't want to do that. The other is what I ended up doing manually by trying to get this to work.

In my searches I found this page: http://sourcey.com/middleman-automated-github-pages-deployments-with-rake

Based on that I made a slight adjustment:
https://github.com/rpbaptist/richardbaptist.nl/blob/master/Rakefile

Basically this rake task recreates the git repo in the `build` directory after each build. It feels excessive, but it's the only thing that works easy and consistently for me.
